src/validator.cpp

```diff
@@ -1066,15 +1100,38 @@ void Validator::ValidatorImpl::validateEquivalenceUnits(const ModelPtr &model, c
                 alreadyReported.push_back(pair);
                 auto unitsName = variable->units() == nullptr ? "" : variable->units()->name();
                 auto equivalentUnitsName = equivalentVariable->units() == nullptr ? "" : equivalentVariable->units()->name();
-                ErrorPtr err = Error::create();
-                err->setDescription("Variable '" + variable->name() + "' has units of '" + unitsName + "' and an equivalent variable '" + equivalentVariable->name() + "' with non-matching units of '" + equivalentUnitsName + "'. The mismatch is: " + hints);
+                IssuePtr err = Issue::create();
+                err->setDescription("Variable '" + variable->name() + "' in component '" + parent1->name() + "' has units of '" + unitsName + "' and an equivalent variable '" + equivalentVariable->name() + "' in component '" + parent2->name() + "' with non-matching units of '" + equivalentUnitsName + "'. The mismatch is: " + hints);
                 err->setModel(model);
-                err->setKind(Error::Kind::UNITS);
-                err->setRule(SpecificationRule::MAP_VARIABLES_IDENTICAL_UNIT_REDUCTION);
-                mValidator->addError(err);
+                err->setCause(Issue::Cause::UNITS);
+                err->setRule(ReferenceRule::MAP_VARIABLES_IDENTICAL_UNIT_REDUCTION);
+                mValidator->addIssue(err);
+            }
+        } else if (multiplier != 0.0) {
+            VariablePair reversePair = std::make_pair(equivalentVariable, variable);
+            auto it = std::find(alreadyReported.begin(), alreadyReported.end(), reversePair);
+            if (it == alreadyReported.end()) {
+                VariablePair pair = std::make_pair(variable, equivalentVariable);
+                ComponentPtr parent1 = std::dynamic_pointer_cast<Component>(variable->parent());
+                ComponentPtr parent2 = std::dynamic_pointer_cast<Component>(equivalentVariable->parent());
+                alreadyReported.push_back(pair);
+                auto unitsName = variable->units() == nullptr ? "" : variable->units()->name();
+                auto equivalentUnitsName = equivalentVariable->units() == nullptr ? "" : equivalentVariable->units()->name();
+                IssuePtr err = Issue::create();
+                std::string des = "Variable '" + variable->name() + "' in component '" + parent1->name() + "' has units of '" + unitsName;
+                des += "' and an equivalent variable '" + equivalentVariable->name() + "' in component '" + parent2->name();
+                des += "' has compatible but not equivalent units of '" + equivalentUnitsName + "'. The mismatch is: " + hints;
+                err->setModel(model);
+                err->setDescription(des);
+                err->setCause(Issue::Cause::UNITS);
+                err->setRule(ReferenceRule::MAP_VARIABLES_IDENTICAL_UNIT_REDUCTION);
+                err->setLevel(Issue::Level::WARNING);
+                mValidator->addIssue(err);
```

Comment:
What is this doing in this PR?

Reply:
It's the one and only example of a warning in the code ... once this PR goes through we can go through and add them throughout, but I figured we needed one to get it started.  I can comment it out for now if you'd rather?

Reply:
I would indeed remove it for now. This is part of step 2: add warnings/hints wherever we feel some should be added.

Reply:
OK, I'll comment it out, and the related tests too.